common/lib/xmodule/xmodule/partitions/partitions.py

```diff
@@ -161,13 +171,16 @@ def from_json(value):
         if value["version"] == 1:
             # If no scheme was provided, set it to the default ('random')
             scheme_id = UserPartition.VERSION_1_SCHEME
-        elif value["version"] == UserPartition.VERSION:
+        elif value["version"] in [2, UserPartition.VERSION]:
             if "scheme" not in value:
                 raise TypeError("UserPartition dict {0} missing value key 'scheme'".format(value))
+
             scheme_id = value["scheme"]
         else:
             raise TypeError("UserPartition dict {0} has unexpected version".format(value))
```

Comment:
It looks like we will need to deploy a patch so that the code in production is forwards-compatible.  We will need to do that before we can safely deploy version 3, in case we need to roll back.